src/commondao/common.dao.ts

```diff
@@ -737,6 +755,7 @@ export class CommonDao<
   async deleteById(id?: string, opt: CommonDaoOptions = {}): Promise<number> {
     if (!id) return 0
     this.requireWriteAccess()
+    if (opt.overrideImmutability !== true) this.requireObjectMutability()
```

Comment:
```suggestion
    if (!opt.overrideImmutability) this.requireObjectMutability()
```

Reply:
Nitpick: because I see this snippet repeated multiple times, you can actually DRY it to `this.requireObjectMutability(opt)` and do an `if` inside the method.

UPD: actually not, since you use this method in places that don't support `overrideImmutability`. Fine to keep then.